jwst/outlier_detection/outlier_detection.py

```diff
@@ -131,81 +226,94 @@ def build_suffix(self, **pars):
         log.debug("Defined output product suffix as: {}".format(
             self.resample_suffix))
 
-    def do_detection(self):
-        """Flag outlier pixels in DQ of input images."""
-        self._convert_inputs()
-        self.build_suffix(**self.outlierpars)
 
-        pars = self.outlierpars
+    @abstractmethod
+    def do_detection(self):
+        pass
+
+
+    @modelwise_operation
+    def compute_weight_threshold(self, resampled, maskpt):
+
+        weight = resampled.wht
+        # necessary in order to assure that mask gets applied correctly
+        if hasattr(weight, '_mask'):
+            del weight._mask
+        mask_zero_weight = np.equal(weight, 0.)
+        mask_nans = np.isnan(weight)
+        # Combine the masks
+        weight_masked = np.ma.array(weight, mask=np.logical_or(
+            mask_zero_weight, mask_nans))
+        # Sigma-clip the unmasked data
+        weight_masked = sigma_clip(weight_masked, sigma=3, maxiters=5)
+        mean_weight = np.mean(weight_masked)
+        # Mask pixels where weight falls below maskpt percent
+        weight_threshold = mean_weight * maskpt
+
+        return weight_threshold
+    
+
+    @timewise_operation
```

Comment:
Perhaps `sectionwise` since (if I'm following the refactor) this would be called for non-tso data?

Reply:
Used for both for the function that computes the median.  For non-TSO data it operates on the zeroth axis, which is typically number of integrations, i.e. time axis, if I understand correctly.  So it would still be an operation that requires the time axis to stay intact.

All of that being said, I think sectionwise works just fine.